extension/cgroupruntimeextension/integration_test.go

```diff
@@ -63,6 +66,21 @@ func cgroupMaxCpu(filename string) (quota int64, period uint64, err error) {
 	return quota, period, err
 }
 
+func startMockECSServer() *httptest.Server {
```

Comment:
Does the integration test interact with this server? Is it a dependency for `gomaxecs.IsECS()`?

Reply:
For the gomaxecs library yes, it is making a http call to the ECS metadata uri.

Does it truly interacts ? I am not sure even for the github action CI, see [my comment in the integration test PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36617#issuecomment-2557877397).

Commands launched in Github action CI:

```bash
gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -skip Sudo -tags=integration,""
gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -skip Sudo -exec sudo -run Sudo -tags=integration,""
```

On my local workstation it is Skipped, compared to the github ci, I removed the `-skip Sudo` flag, the output:

```
$ cd extension/cgroupruntimeextension
$ ../../.tools/gotestsum --rerun-fails=1 --packages="./..." -- -race -timeout 360s -parallel 4 -tags=integration,""
∅  internal/metadata
✓  . (1.056s)

=== Skipped
=== SKIP: . TestCgroupV2SudoIntegration (0.00s)
    integration_test.go:48: System running in hybrid or cgroupv1 mode

DONE 15 tests, 1 skipped in 5.366s
```

I could run the tests in a docker container with some cgroups, open to any suggestion for making the integration test pass.

Reply:
Could you help me identify where the gomaxecs library does the http call to the ECS metadata URI? I quickly checked the `IsECS` method and brought me to this [GetECSMetadataURI](https://github.com/rdforte/gomaxecs/blob/main/internal/config/config.go#L62) method, which just looks for the environment variable.

> I could run the tests in a docker container with some cgroups, open to any suggestion for making the integration test pass.

Nice catch, just pushed this small fix to make them run https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36941

It seems that a changelog file is missing (`make ch-new`) and the linting CI is complaining. Could you please take a look?

Thanks!

Reply:
Hi

1. I rebased the main branch
2. Added a changelog entry

The http call is not made using `IsECS()` but within `gomaxecs.Set()` function call.

I am wondering to revert the current integration test proposition because it is bad implementation.

Proposition is to create a dedicated integration test because it needs to make the http call mocked in order to do a proper integration test. Let me know, multiple options available.

Reply:
Maybe adding a few more test cases (e.g. no container CPU limit defined) would help to understand the behavior, but this approach already looks good to me. Looks very similar to what `gomaxecs` is currently doing: https://github.com/rdforte/gomaxecs/blob/main/internal/task/task_test.go#L455

Reply:
Just pushed a change https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36920/commits/6212bdef7a0c2476b682e1fe3e18e5d13e088f04